### ISSUE TITLE
Retrieve real light capabilities instead of relying on model names

### ISSUE THREAD

**Issue by rogierhofboer** [AlCalzone/node-tradfri-client#69]
At the moment the capabilities of a light are 'hard coded' and based on the model type.
Is it possible to retrieve the real capabilities from the Tradfri gateway?
I have some Hue LCT010 lamps and they do work with the IKEA Tradfri Bridge (I can set the color or the light temperature) , but are not recognized as extended color lamps by node-tradfri-client, because it doesn't know the model. We could add all the Hue lights, but retrieving the capabilities would be more future proof (and there are Zigbee lamps from INNR, Trust, Osram, etc.. as well)

Also none, white, and rgb as spectrum settings do not seem to cover all the possibilities.

**On/off** light (ZigBee Device ID: 0x0000), supports groups, scenes and on/off control

**Dimmable** light (ZigBee Device ID: 0x0100), which supports groups, scenes, on/off and dimming.

**Color temperature** light (ZigBee Device ID: 0x0220), which supports groups, scenes, on/off, dimming, and setting of a color temperature.

**Color light** (ZigBee Device ID: 0x0200), which supports groups, scenes, on/off, dimming and color control (hue/saturation, enhanced hue, color loop and XY)

**Extended Color** light (ZigBee Device ID: 0x0210), same as Color light, but which supports additional setting of color temperature

**Comment by rogierhofboer** [comment]
I will look into this tonight (CET).

**Comment by rogierhofboer** [comment]
**Tradfri White Spectrum**
{"9001":"Woonkamer plafond","9054":0,"9002":1515100956,"9020":1523483501,"9003":65537,"5750":2,"9019":1,"3":{"0":"IKEA of Sweden","1":"TRADFRI bulb E27 WS opal 980lm","2":"","3":"1.2.217","6":1},"3311":[{"5709":24933,"5850":0,"5851":150,"5717":0,"5711":250,"5710":24691,"5706":"f5faf6","9003":0}]}

**Hue Extended Color**
{"9001":"Slaapkamer plafond","9054":0,"9002":1521844374,"9020":1523560973,"9003":65563,"5750":2,"9019":1,"3":{"0":"Philips","1":"LCT010","2":"","3":"1.29.0_r21169","6":1},"3311":[{"5709":29972,"5708":42478,"5850":1,"5851":254,"5717":0,"5711":366,"5707":5438,"5710":26871,"5706":"f1e0b5","9003":0}]}

**Comment by AlCalzone** [comment]
Sounds logical at a first glance. I'll have to look at responses from Tradfri RGB lights to see which properties they report, but I guess they behave like the Hue bulbs. Feel free to provide a PR as I won't have any time to do this myself until some time next week.

Regarding your remarks:
* As far as I understand, all lights with a changeable color (white spectrum or RGB) also support:
  * XY color (although I removed that from the sent payload because I used it as a crutch to set color temp, hue and saturation on earlier gateway firmwares)
  * HEX color (internally converted to hue/sat or color temp)
* RGB could be supported the same way as HEX
* I'd like to keep the spectrum property. Hue and saturation always come in pairs and have to be set in pairs or the gateway does weird shit. So in reality there are only 3 different kinds of bulbs: fixed color, white spectrum and rgb. I'll keep it that way until someone proves that wrong ;)

I think it would be nice to use a declarative approach to the spectrum detection using a `@decorator`.
